src/pages/new-lesson/NewLesson.tsx

```diff
@@ -1,5 +1,142 @@
+import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import AppButton from '~/components/app-button/AppButton'
+
+import { ResourceService } from '~/services/resource-service'
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { AxiosResponse } from 'axios'
+
+import {
+  validations,
+  initialValues,
+  initialFileValue,
+  myResourcesPath
+} from '~/pages/new-lesson/NewLesson.constants'
+import { snackbarVariants } from '~/constants'
+import {
+  ButtonTypeEnum,
+  ButtonVariantEnum,
+  ComponentEnum,
+  ErrorResponse,
+  SizeEnum,
+  NewLessonData,
+  TextFieldVariantEnum
+} from '~/types'
+import { styles } from '~/pages/new-lesson/NewLesson.styles'
+
 const NewLesson = () => {
-  return <h1>New Lesson page</h1>
+  const { t } = useTranslation()
+  const { setAlert } = useSnackBarContext()
+  const navigate = useNavigate()
+
+  const handleResponseError = (error: ErrorResponse) => {
+    setAlert({
+      severity: snackbarVariants.error,
+      message: error ? `errors.${error.code}` : ''
+    })
+  }
+
+  const handleResponse = () => {
+    setAlert({
+      severity: snackbarVariants.success,
+      message: 'newLesson.successMessage'
+    })
+    navigate('/my-resources')
+  }
+
+  const addLesson = (): Promise<AxiosResponse> => {
+    return ResourceService.addLesson(data)
+  }
+
+  const { fetchData } = useAxios({
+    service: addLesson,
+    fetchOnMount: false,
+    defaultResponse: null,
+    onResponse: handleResponse,
+    onResponseError: handleResponseError
+  })
+
+  const handleSave = (content: string) => {
+    console.log('Saved content:', content)
+  }
+
+  const { data, errors, handleInputChange, handleSubmit } =
+    useForm<NewLessonData>({
+      initialValues,
+      validations,
+      onSubmit: fetchData,
+      submitWithData: true
+    })
+  return (
+    <PageWrapper>
+      <Box
+        component={ComponentEnum.Form}
+        onSubmit={handleSubmit}
+        sx={styles.root}
+      >
+        <AppTextField
+          InputLabelProps={styles.titleLabel}
+          InputProps={styles.titleInput}
+          errorMsg={t(errors.title)}
+          fullWidth
+          inputProps={styles.input}
+          label={data.title ? '' : t('newLesson.labels.title')}
+          onChange={handleInputChange('title')}
+          value={data.title}
+          variant={TextFieldVariantEnum.Standard}
+        />
+        <AppTextField
+          InputLabelProps={styles.descriptionLabel}
+          InputProps={styles.descriptionInput}
+          errorMsg={t(errors.description)}
+          fullWidth
+          inputProps={styles.input}
+          label={data.description ? '' : t('newLesson.labels.description')}
+          onChange={handleInputChange('description')}
+          value={data.description}
+          variant={TextFieldVariantEnum.Standard}
+        />
+        <Divider sx={styles.divider} />
+        <Editor
```

Comment:
Maybe, It is better to create component for this editor

Reply:
yeap, it makes sense. Fixed